### PR TITLE
YLP-1590 change role with auth0

### PR DESCRIPTION
**Pull request by fatb38**

### CODE REVIEW

File: packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    // Second dialog (privacy and terms)
+    const consentDialog = screen.getByTestId('switch-role-consent-dialog')
+    expect(consentDialog).toBeVisible()
+    const privacyCheckbox = within(consentDialog).getByLabelText('Privacy policy checkbox')
+    const termsCheckbox = within(consentDialog).getByLabelText('Terms checkbox')
+    const feedbackCheckbox = within(consentDialog).getByLabelText('Feedback checkbox')
+    const acceptButton = within(consentDialog).getByRole('button', { name: 'Accept' })
+    expect(within(consentDialog).getByTestId('switch-role-consent-dialog-label-privacy-policy')).toBeVisible()
+    expect(within(consentDialog).getByTestId('switch-role-consent-dialog-label-terms')).toBeVisible()
+    expect(within(consentDialog).getByTestId('switch-role-consent-dialog-label-feedback')).toBeVisible()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx:152]:
```suggestion
    expect(within(consentDialog).getByTestId('switch-role-consent-dialog-label-privacy-policy')).toHaveTextContent('I have read and accepted YourLoops Privacy Policy. Hereby, I consent to the processing of my personal data by Diabeloop so I can benefit from YourLoops services.')
    expect(within(consentDialog).getByTestId('switch-role-consent-dialog-label-terms')).toHaveTextContent('I have read and accepted YourLoops Terms of use. I confirm that I am an authorized healthcare practitioner according to local laws and regulation. I undertake to use YourLoops solely in accordance with these Terms of Use and what is allowed by the law.')
    expect(within(consentDialog).getByTestId('switch-role-consent-dialog-label-feedback')).toHaveTextContent('I agree to receive information and news from Diabeloop. (optional)')
    const privacyCheckbox = within(consentDialog).getByLabelText('Privacy policy checkbox')
    const termsCheckbox = within(consentDialog).getByLabelText('Terms checkbox')
    const feedbackCheckbox = within(consentDialog).getByLabelText('Feedback checkbox')
    const acceptButton = within(consentDialog).getByRole('button', { name: 'Accept' })
```

File: packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    expect(privacyCheckbox).toBeVisible()
+    expect(termsCheckbox).toBeVisible()
+    expect(feedbackCheckbox).toBeVisible()
+    expect(acceptButton).toBeDisabled()
+    await userEvent.click(privacyCheckbox)
+    await userEvent.click(termsCheckbox)
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx:158]:
```suggestion
    expect(acceptButton).toBeDisabled()
    await userEvent.click(termsCheckbox)
    expect(acceptButton).toBeEnabled()
```

File: packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    await userEvent.click(feedbackCheckbox)
+    await userEvent.click(acceptButton)
+
+    // Third dialog (HCP profession)
+    const professionDialog = screen.getByTestId('switch-role-profession-dialog')
+    const hcpProfessionSelect = within(professionDialog).getByTestId('dropdown-profession-selector')
+    const validateButton = within(professionDialog).getByRole('button', { name: 'Validate' })
+    expect(professionDialog).toBeVisible()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx:166]:
```suggestion
```

File: packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    expect(hcpProfessionSelect).toBeVisible()
+    expect(validateButton).toBeDisabled()
+    expect(hcpProfessionSelect.textContent).toEqual('​')
+
+    fireEvent.mouseDown(within(screen.getByTestId('dropdown-profession-selector')).getByRole('button'))
+    fireEvent.click(screen.getByRole('option', { name: 'Dietitian' }))
+    expect(hcpProfessionSelect).toHaveTextContent('Dietitian')
+    await userEvent.click(validateButton)
+    expect(changeUserRoleToHcpMock).toHaveBeenCalled()
+  })
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx:176]: optional: test the cancel and decline buttons

File: packages/yourloops/components/switch-role/index.tsx [hunk 5]
```diff
@@ -74,18 +76,9 @@ function SwitchRoleDialogs(props: SwitchRoleDialogsProps): JSX.Element {
     setSwitchRoleStep(SwitchRoleToHcpSteps.profession)
   }
 
-  React.useEffect(() => {
-    if (props.open && switchRoleStep === SwitchRoleToHcpSteps.none) {
```
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/components/switch-role/index.tsx:78]: 👽

File: packages/yourloops/components/menus/user-menu.tsx [hunk 6]
```diff
@@ -134,7 +134,7 @@ function UserMenu(): JSX.Element {
         maxWidth={250}
         onClick={event => setAnchorEl(event.currentTarget)}
       >
-        <Box id="user-role-icon" display="flex">
+        <Box data-testid={`${user.role}-icon`} display="flex">
```
Comment (TimCanu) [inline packages/yourloops/components/menus/user-menu.tsx:137]: I suggest keeping `user-role-icon` as a data-testid. I don't see the value in having the role in the data-testid (except making it more difficult to write E2E tests)

File: locales/en/yourloops.json [hunk 7]
```diff
@@ -341,6 +341,7 @@
   "signup-steppers-ending-message-2": "If you don’t receive any email from us, please contact our support at <a href=\"mailto:technical.support@diabeloop.com\">technical.support@diabeloop.com</a>",
   "signup-unknown-step": "Unknown step",
   "ssn": "SSN",
+  "switch-role-success": "Your account has been successfully switched to HCP",
```
Comment (TimCanu) [inline locales/en/yourloops.json:344]: You fixed the trad but not the tests using it!

**Review by TimCanu — APPROVED** [review]
Perfecto :pinched_fingers:

**Pull request merged**